bitar/src/api/compress.rs

```diff
@@ -0,0 +1,269 @@
+use std::collections::HashMap;
+use std::default::Default;
+use std::error;
+use std::fmt;
+use std::path::PathBuf;
+
+use blake2::{Blake2b512, Digest};
+use futures_util::{future, StreamExt};
+use tokio::fs;
+use tokio::io;
+use tokio::io::AsyncSeekExt;
+use tokio::io::{AsyncRead, AsyncWrite, AsyncWriteExt};
+
+use crate::chunk_dictionary;
+use crate::chunker;
+use crate::Compression;
+use crate::HashSum;
+
+pub const PKG_VERSION: &str = env!("CARGO_PKG_VERSION");
+
+/// Options for the `create_archive` function
+#[derive(Clone, Debug)]
+pub struct CreateArchiveOptions {
+    /// The configuration to use when creating a chunk stream from the input
+    pub chunker_config: chunker::Config,
+
+    /// Number of parallel buffers to use when manipulating chunks
+    pub num_chunk_buffers: usize,
+
+    /// The length that the source hash should be truncated to for the output
+    pub source_hash_length: usize,
```

Comment:
Think this should be `chunk_hash_length` since it's for truncating the length of each chunk's hash.
The hash of the full source file is always stored with 64 bytes of length.

Reply:
Indeed `chunk_hash_length` is the correct name, made the changes